torch_geometric/transforms/add_metapaths.py

```diff
@@ -119,26 +126,31 @@ def __call__(self, data: HeteroData) -> HeteroData:
                     edge_type) in edge_types, f"'{edge_type}' not present"
 
             edge_type = metapath[0]
+            adj1_edge_attr = self._get_edge_attr(data, edge_type)
             adj1 = SparseTensor.from_edge_index(
                 edge_index=data[edge_type].edge_index,
-                sparse_sizes=data[edge_type].size())
+                sparse_sizes=data[edge_type].size(), edge_attr=adj1_edge_attr)
 
             if self.max_sample is not None:
                 adj1 = self.sample_adj(adj1)
 
             for i, edge_type in enumerate(metapath[1:]):
+                adj2_edge_attr = self._get_edge_attr(data, edge_type)
                 adj2 = SparseTensor.from_edge_index(
                     edge_index=data[edge_type].edge_index,
-                    sparse_sizes=data[edge_type].size())
+                    sparse_sizes=data[edge_type].size(),
+                    edge_attr=adj2_edge_attr)
 
                 adj1 = adj1 @ adj2
 
                 if self.max_sample is not None:
                     adj1 = self.sample_adj(adj1)
 
-            row, col, _ = adj1.coo()
+            row, col, value = adj1.coo()
```

Comment:
how about calling it `weights` to be consistent with the argument name `weighted`